### ISSUE TITLE
Segmentation fault at "Go to page" button, only in specific book + saved words bug

### ISSUE THREAD

**Issue by Szybet** [Kobo-InkBox/inkbox#27]
As in the tittle. I can provide the book on pv if you want to test it. Here is the GUI_DEBUG log:
```
kobo:/boot/flags# iniparser: cannot open /mnt/onboard/.kobo/nightmode.ini
cat: can't open '/mnt/onboard/.kobo/Kobo/Kobo eReader.conf': No such file or directory
21/04/2022 @ 22:38:10 | main: Running on device n306
21/04/2022 @ 22:38:11 | main: Device has Wi-Fi capabilities
21/04/2022 @ 22:38:26 | MainWindow: Checking for available updates
21/04/2022 @ 22:38:26 | MainWindow: checkForUpdate: No update available.
21/04/2022 @ 22:38:27 | MainWindow: Showing USB Mass Storage dialog
21/04/2022 @ 22:38:28 | MainWindow: Setting initial brightness to 0
21/04/2022 @ 22:38:28 | functions: cinematicBrightness: Setting brightness to 0
21/04/2022 @ 22:38:35 | generalDialog: Cancel button clicked
21/04/2022 @ 22:38:46 | MainWindow: Launching Reader Framework
Numeric mode unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
21/04/2022 @ 22:39:02 | reader: Opening file '/mnt/onboard/onboard/fun-fun/Andrzej Ziemia?ski - Cyberpunk. Odrodzenie.pdf'
mount: mounting /opt/config/06-saved_words on /kobo/mnt/onboard/.adds/inkbox/.config/06-saved_words failed: No such file or directory
21/04/2022 @ 22:39:03 | reader: Setting local settings environment up
21/04/2022 @ 22:39:03 | reader: Local Reading Settings: Setting brightness to 0
21/04/2022 @ 22:39:03 | functions: cinematicBrightness: Setting brightness to 0
21/04/2022 @ 22:39:03 | reader: Setting font to 'Source Serif Pro'
21/04/2022 @ 22:39:03 | reader: Setting night mode to OFF
21/04/2022 @ 22:39:03 | reader: Book file format: PDF
21/04/2022 @ 22:39:03 | reader: Setting page refresh to each 3 pages
21/04/2022 @ 22:39:03 | reader: Book file format: PDF
21/04/2022 @ 22:39:03 | reader: Getting viewport orientation for PDF file '/mnt/onboard/onboard/fun-fun/Andrzej Ziemia?ski - Cyberpunk. Odrodzenie.pdf'
21/04/2022 @ 22:39:14 | reader: getPdfOrientation: Orientation is portrait
21/04/2022 @ 22:39:14 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:14 | functions: defineDefaultPageSizeDefined default PDF page width to 708
21/04/2022 @ 22:39:19 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:19 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:20 | reader: Setting font size to 10 points
21/04/2022 @ 22:39:20 | functions: defineDefaultPageSize: Defined default ePUB page height to 450
21/04/2022 @ 22:39:20 | functions: defineDefaultPageSize: Defined default ePUB page width to 450
21/04/2022 @ 22:39:21 | reader: Total PDF pages number: 295
21/04/2022 @ 22:39:26 | reader: 'Next' button clicked: going to page 2
21/04/2022 @ 22:39:26 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:26 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:28 | reader: 'Next' button clicked: going to page 3
21/04/2022 @ 22:39:28 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:28 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:30 | reader: 'Next' button clicked: going to page 4
21/04/2022 @ 22:39:30 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:30 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:31 | reader: 'Next' button clicked: going to page 5
21/04/2022 @ 22:39:31 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:31 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:33 | reader: 'Next' button clicked: going to page 6
21/04/2022 @ 22:39:33 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:33 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:35 | reader: 'Next' button clicked: going to page 7
21/04/2022 @ 22:39:35 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:35 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:36 | reader: 'Next' button clicked: going to page 8
21/04/2022 @ 22:39:36 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:36 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:38 | reader: 'Next' button clicked: going to page 9
21/04/2022 @ 22:39:38 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:38 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
21/04/2022 @ 22:39:42 | reader: 'Options' button clicked
21/04/2022 @ 22:39:42 | reader: Showing menu bar
21/04/2022 @ 22:39:44 | reader: Showing 'Go to page' dialog
21/04/2022 @ 22:39:47 | generalDialog: OK button clicked
21/04/2022 @ 22:39:47 | reader: Going to page 90
21/04/2022 @ 22:39:47 | functions: defineDefaultPageSizeDefined default PDF page height to 974
21/04/2022 @ 22:39:47 | functions: defineDefaultPageSizeDefined default PDF page width to 708
QIODevice::read (QFile, "/inkbox/book/split/."): device not open
QIODevice::read (QFile, "/inkbox/book/split/.."): device not open
Segmentation fault
```
Also for this line, why saved words doesn't work for me:
```
mount: mounting /opt/config/06-saved_words on /kobo/mnt/onboard/.adds/inkbox/.config/06-saved_words failed: No such file or directory
```
Once you said to format the usb storage becouse something doesn't worked, becouse i was tinkering too much. other config files recreated itself but `saved_words` doesn't

**Comment by tux-linux** [comment]
I got the saved words bug too here for some reason. Will try to dig and see what causes it.
As for PDF, we need to dig deep into Qt's internals to understand why there is a segmentation fault for no apparent reason...

**Comment by tux-linux** [comment]
Fixed the PDF issue in 5a38ecdab3b4a198ec3cb709437ac03767f92f01

**Comment by Szybet** [comment]
I will test it and then close the issue if the problem is resolved
